src/MOI_wrapper.jl

```diff
@@ -332,3 +336,72 @@ function LQOI.delete_sos!(model::Optimizer, begin_index::Int, end_index::Int)
     c_api_delsos(model.inner, begin_index - 1, end_index - 1)
     return
 end
+
+
+function scalediagonal!(V, I, J, scale)
+    #  LQOI assumes 0.5 x' Q x, but CPLEX requires the list of terms, e.g.,
+    #  2x^2 + xy + y^2, so we multiply the diagonal of V by 0.5. We don't
+    #  multiply the off-diagonal terms since we assume they are symmetric and we
+    #  only need to give one.
+    #
+    #  We also need to make sure that after adding the constraint we un-scale
+    #  the vector because we can't modify user-data.
+    for i in 1:length(I)
+        if I[i] == J[i]
+            V[i] *= scale
+        end
+    end
+    return
+end
+
+
+function LQOI.set_quadratic_objective!(model::Optimizer, I::Vector{Int}, J::Vector{Int}, V::Vector{Float64})
+    @assert length(I) == length(J) == length(V)
+    #scalediagonal!(V, I, J, 0.5)
+    CPLEX.add_qpterms!(model.inner, I, J, V)
+    #scalediagonal!(V, I, J, 2.0)
+    return
+end
+
+LQOI.solve_quadratic_problem!(model::Optimizer) = LQOI.solve_linear_problem!(model)
+
+function LQOI.get_quadratic_primal_solution!(model::Optimizer, dest)
+    c_api_getxqxax(model.inner, dest)
+    return
+end
+
+function LQOI.get_quadratic_dual_solution!(model::Optimizer, dest)
+    c_api_getqconstrslack(model.inner, dest)
+    return
+end
+
+function LQOI.add_quadratic_constraint!(model::Optimizer,
+        affine_columns::Vector{Int}, affine_coefficients::Vector{Float64},
+        rhs::Float64, sense::Cchar,
+        I::Vector{Int}, J::Vector{Int}, V::Vector{Float64})
+    @assert length(I) == length(J) == length(V)
+    scalediagonal!(V, I, J, 0.5)
```

Comment:
Are these needed? The CPLEX docs aren't clear: https://www.ibm.com/support/knowledgecenter/SSSA5P_12.6.2/ilog.odms.cplex.help/refcallablelibrary/socpapi/addqconstr.html

Maybe that's the cause of the failures?

Reply:
I tested with and without scaling in constraints. I have more tests passing with scaling than without. Moreover the JuMP example in the first message of the PR does not return appropriate values without the scaling.

Reply:
Scaling here seems appropriate, since `add_qconstr!` assumes a list of terms:
https://github.com/JuliaOpt/CPLEX.jl/blob/bb3f45c236fb0afb2fa2ed75c448e17b9a64a61e/src/CplexSolverInterface.jl#L298-L299
https://mathprogbasejl.readthedocs.io/en/latest/lpqcqp.html#addquadconstr!